app/screens/global_threads/threads_list/thread/thread_footer/thread_footer.tsx

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useMemo} from 'react';
+import {View} from 'react-native';
+
+import FormattedText from '@components/formatted_text';
+import UserAvatarsStack from '@components/user_avatars_stack';
+import {changeOpacity, makeStyleSheetFromTheme} from '@utils/theme';
+
+import type ThreadModel from '@typings/database/models/servers/thread';
+import type UserModel from '@typings/database/models/servers/user';
+
+export type Props = {
+    author?: UserModel;
+    currentUserId: string;
+    participants: UserModel[];
+    teammateNameDisplay: string;
+    testID: string;
+    thread: ThreadModel;
+    theme: Theme;
+};
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        container: {
+            flexDirection: 'row',
+            alignItems: 'center',
+            minHeight: 40,
+        },
+        avatarsContainer: {
+            marginRight: 12,
+            paddingVertical: 8,
+        },
+        replyIconContainer: {
+            top: -1,
+            marginRight: 5,
+        },
+        replies: {
+            alignSelf: 'center',
+            color: changeOpacity(theme.centerChannelColor, 0.64),
+            fontSize: 12,
+            fontWeight: '600',
+            marginRight: 12,
+        },
+        unreadReplies: {
+            alignSelf: 'center',
+            color: theme.sidebarTextActiveBorder,
+            fontSize: 12,
+            fontWeight: '600',
+            marginRight: 12,
+        },
+    };
+});
+
+const ThreadFooter = ({author, currentUserId, participants, teammateNameDisplay, testID, theme, thread}: Props) => {
+    const style = getStyleSheet(theme);
+
+    let repliesComponent;
+    if (thread.unreadReplies) {
+        repliesComponent = (
+            <FormattedText
+                id='threads.newReplies'
+                defaultMessage='{count} new {count, plural, one {reply} other {replies}}'
+                style={style.unreadReplies}
+                testID={`${testID}.unread_replies`}
+                values={{
```

Comment:
This probably could use some `useMemo` to avoid re-renders.

Short term it won't be doing much, since a re-render of the parent component will do a re-render on the child, and it seems that `/components/formatted_text/index.tsx` is not using any kind of memoization (not even in terms of callbacks or effects). Nevertheless, I find it a good practice.